src/DssSpell.t.sol

```diff
@@ -2089,39 +2089,41 @@ contract DssSpellTest is DSTest, DSMath {
 
         checkCollateralValues(afterSpell);
     }
-/*
-    function test_withdraw_ceiling() public {
-        vote(address(spell));
-        spell.schedule();
-        hevm.warp(spell.nextCastTime());
-        spell.cast();
-        assertTrue(spell.done());
 
-        bytes32 ilk = bytes32("UNIV2DAIUSDC-A");
-        jug.drip(ilk);
-
-        // Confirm pip value.
-        //DSValueAbstract pip = DSValueAbstract(addr.addr("PIP_UNIV2DAIUSDC_A"));
-        //assertEq(uint256(pip.read()), 22_495_725 * WAD);
-
-        // Confirm Vat.ilk.spot value.
-        (uint256 Art, uint256 rate, uint256 spot, uint256 line,) = vat.ilks(ilk);
-        //assertEq(spot, 21_424_500 * RAY);
-
-        // Test that a draw can be performed.
-        address urn = addr.addr("RWA002_A_URN");
-        giveAuth(urn, address(this));
-        RwaUrnLike(urn).hope(address(this));  // become operator
-        uint256 room = sub(line, mul(Art, rate));
-        uint256 drawAmt = room / RAY;
-        if (mul(divup(mul(drawAmt, RAY), rate), rate) > room) {
-            drawAmt = sub(room, rate) / RAY;
+    function testRwaDrawUpToDebtCeiling() public {
+        giveAuth(address(vat), address(this));
+        vat.file("Line", uint256(-1));  // ensure the global debt ceiling doesn't interfere with draws
+        bytes32[] memory ilks = reg.list();
+        for(uint256 i = 0; i < ilks.length; i++) {
+            bytes32 ilk = ilks[i];
+            if (reg.class(ilk) != 3) continue;
```

Comment:
Currently this is the convention for the "class" in ilk registry: 
`Classification code (1 - clip, 2 - flip, 3+ - other)`
In the future there might be a situation where the class is 3 but this is not a RWA type of collateral. 
In that case this code will probably end up not working.

Can consider replacing this line with:
` if (compareStrings(substring(reg.symbol(ilk), 0, 3), "RWA")) continue;`

and using these helpers:
    function compareStrings(string memory a, string memory b) public pure returns (bool) {
        return (keccak256(abi.encodePacked((a))) == keccak256(abi.encodePacked((b))));
    }

`
function substring(string memory str, uint startIndex, uint endIndex) public pure returns (string memory) {
        bytes memory strBytes = bytes(str);
        bytes memory result = new bytes(endIndex-startIndex);
        for(uint i = startIndex; i < endIndex; i++) {
            result[i-startIndex] = strBytes[i];
        }
        return string(result);
    }
`

Not a must imo.

Reply:
My expectation is that class 3 will be exclusively used for MIP21-style RWAs. On the flip side, there's no guarantee that the "RWA" prefix will always apply exclusively to MIP21-style RWAs. So IMO `class == 3` is probably the safest condition for now. Unless someone like @brianmcmichael or @godsflaw knows more about how these conventions are intended to function.

Reply:
Yeah, for now, class 3 is this class of RWA (MIP21 style), and we may see different classes in the future for other types of assets.  I also agree with Kurt, we may see a non RWA prefix, so the "class" is really our only way to tag these reliably.